processor/src/main/java/org/mapstruct/ap/internal/util/NativeTypes.java

```diff
@@ -233,6 +221,185 @@ private boolean floatHasBecomeZero() {
         }
     }
 
+   private static class BooleanAnalyzer implements LiteralAnalyzer {
+
+        @Override
+        public void validate(String s) {
+            if ( !( "true".equals( s ) || "false".equals( s ) ) ) {
+                throw new IllegalArgumentException("only 'true' or 'false' are supported");
+            }
+        }
+
+        @Override
+        public Class<?> getLiteral() {
+            return boolean.class;
+        }
+    }
+
+    private static class CharAnalyzer implements LiteralAnalyzer {
+
+        @Override
+        public void validate(String s) {
+            if ( !(s.length() == 3 && s.startsWith( "'" ) && s.endsWith( "'" )) ) {
+                throw new NumberFormatException( "invalid character literal" );
+            }
+        }
+
+        @Override
+        public Class<?> getLiteral() {
+            return char.class;
+        }
+    }
+
+    private static class ByteAnalyzer implements LiteralAnalyzer {
+
+        @Override
+        public void validate(String s) {
+            NumberRepresentation br = new NumberRepresentation( s, true, false, false ) {
+
+                @Override
+                void parse(String val, int radix) {
+                    Byte.parseByte( val, radix );
+                }
+            };
+            br.validate();
+        }
+
+        @Override
+        public Class<?> getLiteral() {
+            return int.class;
+        }
+    }
+
+    private static class DoubleAnalyzer implements LiteralAnalyzer {
+
+        @Override
+        public void validate(String s) {
+            NumberRepresentation br = new NumberRepresentation( s, false, false, false ) {
+
+                @Override
+                void parse(String val, int radix) {
+                    Double d = Double.parseDouble( radix == 16 ? "0x" + val : val );
+                    if ( doubleHasBecomeZero( d  ) ) {
+                        throw new NumberFormatException( "floating point number too small" );
```

Comment:
Does this disallow `@Mapping(constant = "0.0")` ?

Reply:
It should. I'll add a test case for that one..